### PR TITLE
Added minimal price to configurable product upon fetching via API

### PR DESCRIPTION
**Pull request by rogyar**
Description

Upon getting a configurable product via API (/rest/V1/products/:sku) the product does not contain any useful information about the price. The minimal price is useful here in order to avoid additional requests to the API for getting the configurable product price information.

Fixed Issues

magento/magento2#10644: REST API Configurable Product has price
Manual testing scenarios

Create a configurable product accessible on the storefront
Get the product information via API using GET request to the rest/V1/products/:sku
You should get a minimal price among the custom attributes for the configurable product.

### CODE REVIEW

File: app/code/Magento/Catalog/Model/ProductRepository.php [hunk 1]
```diff
@@ -23,6 +23,7 @@
 use Magento\Framework\Exception\NoSuchEntityException;
 use Magento\Framework\Exception\StateException;
 use Magento\Framework\Exception\ValidatorException;
+use Magento\ConfigurableProduct\Model\Product\Type\Configurable;
```
Comment (orlangur) [inline app/code/Magento/Catalog/Model/ProductRepository.php]: You cannot refer to `ConfigurableProduct` module in `Catalog`.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by rogyar** [comment]
The magento/magento2#10644 issue is fixed only partially in this PR. But I guess this PR can be reviewed as a separate improvement. 
As for the configurable products in the list (related to the mentioned issue), it's quite difficult to have a minimal price only for the configurable products without making "tweaks" like adding minimal price to the collection and then loop through the products from the loaded collection to remove the minimal price from the simple products.

File: app/code/Magento/Catalog/Model/ProductRepository.php (hunk 1 above)
Comment (orlangur) [reply app/code/Magento/Catalog/Model/ProductRepository.php]: As far as I remember I already asked in previous PR but it was force pushed or something.

I don't see a good way to move such addition to `ConfigurableProduct` module and I don't see disadvantages of having `min_price` for simple product (for example, if simple product has non-free custom options, price without any of them would be minimal).

@vrann thoughts?

**Comment by vkublytskyi** [comment]
@rogyar Please fix failed tests

**Review by LSERRE — COMMENTED** [review]
I need it so much

**Comment by rogyar** [comment]
Hello @LSERRE. Unfortunately, at this moment I don't have enough time to fix all tests because many tests are related to this issue and the new behaviour is not acceptable.

**Comment by LSERRE** [comment]
Hello @rogyar. I simply did a plugin in a custom module on app/code/Magento/ConfigurableProduct/Model/Product/Type/Configurable/Price.php. I used your logic so thank you anyway, my problem is resolved.

**Comment by magento-engcom-team** [comment]
Hi @rogyar 
We are closing this Pull Request now due to inactivity.
If you would like to revisit this in the future – please take a look at the failed tests and reopen.

Thank you for collaboration

**Pull request closed without merging**